tests/cypress/integration/filter.spec.js

```diff
@@ -0,0 +1,269 @@
+
+    // Test if application has non-zero subjects
+
+    cy
+      .get(nsInfoTable('table tbody tr:first td:last'))
+      .contains(/[1-9]+[0-9]*\/[1-9]+[0-9]*/);
+
+    // Monitor how many times summary is recalculated vias shiny:recalculated
+    // event
+    cy
+      .get('@filter_summary')
+      .within(() => {
+        cy
+          .get('.shiny-bound-output')
+          .invoke('on', 'shiny:recalculated', cy.stub().as('summary_change'));
+      });
+
+    // Move slider via cypress2 mouse drag
+    // ------------------------------------
+    cy
+      .get('@filter_variables')
+      .within(() => {
+        // Make sure that the overlay plot is rendered
+        cy
+          .get('.filterPlotOverlayRange .shiny-plot-output')
+          .children()
+          .should('have.length.gte', 1);
+
+        cy
+          .get('.filterPlotOverlayRange .shiny-plot-output')
+          .not('.recalculating');
+
+        // Move the handle 3x
+        //  note: this is necessary as developer wasn't able to define a single
+        //    long drag. This is open to improvement
+        cy.get('.irs-handle.from').as('irs-handle');
+
+        const mousedownOpts = {
+          button: 0, which: 1, pageX: 600, pageY: 100,
+        };
+        const mousemoveOpts = {
+          clientX: 300, clientY: 400,
+        };
+
+        cy.get('@irs-handle').trigger('mousedown', 'top', mousedownOpts);
+        cy.get('@irs-handle').trigger('mousemove', 'top', mousemoveOpts);
+        cy.get('@irs-handle').trigger('mouseup', 'top');
+
+        cy
+          .get('.filterPlotOverlayRange .shiny-plot-output')
+          .not('.recalculating');
+        // cy.get('@summary_change', { timeout: 20000 }).should('have.callCount', 1)
+
+        cy.get('@irs-handle').trigger('mousedown', 'top', mousedownOpts);
+        cy.get('@irs-handle').trigger('mousemove', 'top', mousemoveOpts);
+        cy.get('@irs-handle').trigger('mouseup', 'top');
+
+        cy
+          .get('.filterPlotOverlayRange .shiny-plot-output')
+          .not('.recalculating');
+
+        cy.get('@irs-handle').trigger('mousedown', 'top', mousedownOpts);
+        cy.get('@irs-handle').trigger('mousemove', 'top', mousemoveOpts);
+        cy.get('@irs-handle').trigger('mouseup', 'top');
+
+        cy
+          .get('.filterPlotOverlayRange .shiny-plot-output')
+          .not('.recalculating');
+      });
+
+    // Let shiny finish rendering
+    cy
+      .get('html')
+      .not('.shiny-busy');
+
+    // Verify that data is filtered (selected subjets != dataset)
+    // ------------------------------------
+    cy
+      .get(nsInfoTable('table tbody tr:first td:last'), { timeout: 10000 })
+      .should('satisfy', ($el) => {
+        const result = /([0-9]+)\/([0-9]+)/.exec($el[0].innerText);
+        return result.length === 3 && result[1] !== result[2];
+      });
+  });
+
+  //  Active tab
+  // ######################################
+
+  it('Active tab has shiny content rendered', () => {
```

Comment:
How about adding this kind of check to afterEach?

Reply:
Brilliant 🤯🤩

Done!